target/linux/ramips/image/mt76x8.mk

```diff
@@ -18,6 +18,18 @@ define Device/tplink
 endef
 DEVICE_VARS += TPLINK_FLASHLAYOUT TPLINK_HWID TPLINK_HWREV TPLINK_HWREVADD TPLINK_HVERSION
 
+define Device/tplink_c20-v4
+  $(Device/tplink)
+  DTS := ArcherC20v4
+  IMAGE_SIZE := 7808k
+  DEVICE_TITLE := TP-Link ArcherC20 v4
+  TPLINK_FLASHLAYOUT := 8Mmtk
+  TPLINK_HWID := 0xc200004
+  TPLINK_HWREV := 0x283
```

Comment:
In my region firmware HWREV is 0x283. I didn't check a HWREV code for other regions. It turns out that a code everywhere different. Is there options?

Reply:
Look here: https://github.com/lede-project/source/commit/24043a0d2e01b9843c0dc529205b3b0bc7ecbbf9

>Factory image notes:
>
> These devices use version 3 of TP-Link header, fortunately without RSA
signature (at least in case of devices sold in Europe). The difference
lays in the requirement for a non-zero value in "Additional Hardware
Version" field. Ideally, it should match the value stored in vendor
firmware header on device ("0x4"/"0x13" for these devices) but it seems
that anything other than "0" is correct.

You can set a random "non-zero" value in this field and everything is OK

Reply:
But in my case difference only in HWREV code! HWREVADD is the same for EU and RU regions.
![2018-01-11_214328_cr](https://user-images.githubusercontent.com/15726175/34841250-ae3f27c4-f718-11e7-8aec-228c51e7730a.png)